### PR TITLE
Lev 299 User Create page

### PR DESCRIPTION
**Pull request by nBaqai**
- user create page to add user
- added placeholder for team select
- added extra fields validations email and team selection
- added button on Manage organisation page and link with add user page
- added currentTeamId as query param for team to be selected automatically and for using cancel button (user enter in create user page from Manage organisation or Manage team page cancel button will take them where they originally took the action.
- automatically retaining user input on validation error.
- upgraded hmpo repos to eliminate CVE-2024-45590, CVE-2024-45296

### CODE REVIEW

File: controllers/UserManagement/UserCreateController.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      }, { firstname: firstname, lastname: lastname, email: email, teamId: teamIdToAdd });
+
+      req.sessionModel.set('userFullName', firstname + ' ' + lastname);
+      req.sessionModel.unset('firstNameUserEntered');
+      req.sessionModel.unset('lastNameUserEntered');
+      req.sessionModel.unset('emailUserEntered');
+      req.sessionModel.unset('teamIdUserSelected');
+
+    } catch (err) {
+      req.sessionModel.set('userFullName', '');
```
Comment (ashley-bickerstaff-homeoffice) [inline controllers/UserManagement/UserCreateController.js:61]: When I tested this, if the user already exists, it fails... which is correct. But it only fails with "There was a problem adding user". It feels as though this error should be explanatory given I can imagine this scenario occurring.

Additionally, if you enter this error state, it redirects back to the previous page, but then if you go the "Add user" page again, the form is pre-filled with the previous state. It feels like it should be cleared in this catch block

File: views/pages/organisation/team.html [hunk 2]
```diff
@@ -20,6 +21,24 @@
...
+        {% set status = '' %}
+        {% set html = '' %}
+
+        {% if addedUser %}
+          {% if userFullName %}
+            {% set status = "success" %}
+            {% set html = "<h2>You have successfully added " + userFullName %}
+          {% else %}
+            {% set status = "failure" %}
+            {% set html="<h2>There was a problem adding user</h2>" %}
```
Comment (ashley-bickerstaff-homeoffice) [inline views/pages/organisation/team.html:33]: Can we tighten the grammar here?

```suggestion
            {% set html="<h2>There was a problem adding the user</h2>" %}
```

Alternatively, if we reach a dead-end with the "adding a user that already exists" problem I commented on above, we could expand this to "There was a problem adding the user. The user may already exist in LEV". This is something we should validate with Jo first though.

**Review by ashley-bickerstaff-homeoffice — CHANGES_REQUESTED** [review]

File: controllers/UserManagement/UserCreateController.js (hunk 1 above)
Comment (nBaqai) [reply controllers/UserManagement/UserCreateController.js:57]: I have updated the error message as discussed with Jo, and also cleared the pre-filled previous state on error.

File: views/pages/organisation/team.html (hunk 2 above)
Comment (nBaqai) [reply views/pages/organisation/team.html]: The error message is changed as discussed with Jo

**Pull request merged**